napari/_vispy/layers/image.py

```diff
@@ -41,15 +43,20 @@ def __init__(
             texture_format=texture_format,
         )
 
-    def get_node(self, ndisplay: int) -> Node:
+    def get_node(self, ndisplay: int, dtype=None) -> Node:
         # Return custom node if we have one.
         if self._custom_node is not None:
             return self._custom_node
 
         # Return Image or Volume node based on 2D or 3D.
-        if ndisplay == 2:
-            return self._image_node
-        return self._volume_node
+        res = self._image_node if ndisplay == 2 else self._volume_node
+        if (
+            res.texture_format != "auto"
+            and dtype is not None
+            and _VISPY_FORMAT_TO_DTYPE[res.texture_format] != dtype
+        ):
+            raise ValueError("dtype does not match texture_format")
```

Comment:
To raise an exception earlier when someone will next time touch this code will not need to understand vispy error. 

(It also allows to avoid linter complaining about unused parameter)

Reply:
Hmmm. That doesn't feel compelling to me... But maybe we can leave it if we add a comment, e.g.

```suggestion
            # it is a bug to hit this error — it is here to catch bugs
            # early when we are creating the wrong nodes or
            # textures for our data
            raise ValueError("dtype does not match texture_format")
```